tests/conftest.py

```diff
@@ -130,9 +130,31 @@ def teardown():
     _app.jwt_public_keys = {_app.config['USER_API']: {
             'key-test': utils.read_file('resources/keys/test_public_key.pem')
     }}
+
     return _app
 
 
+@pytest.fixture(scope='function')
+def client_toggled(app, request):
+    """
+    Will toggle app config parameter for the test using this and return test client
+
+    USAGE:
+    @pytest.mark.config_toggle(parameter='PARAM', value=VAL)
+    def test_mytest(client_toggled):
+        ...
+
+    NOTE: App config gets reset each time function-scoped app fixture is executed,
+    no need to reset here
+    """
+    params = request.node.get_marker('config_toggle')
+
+    app.config[params.kwargs['parameter']] = params.kwargs['value']
```

Comment:
You're making the assumption that `app` fixture is functionally scoped. If it were to not be, this fixture would have side effects in other tests.... 

Not 100% on that. You could experiment, or just reset the value after the yield.

Reply:
addressed in #106